jsonpickle/util.py

```diff
@@ -16,13 +16,15 @@
 import sys
 import time
 import types
+from collections.abc import Iterator as abc_iterator
 
-from . import compat, tags
-from .compat import abc_iterator, class_types, iterator_types, numeric_types
+from . import tags
 
+NUMERIC_TYPES = (int, float)
+ITERATOR_TYPE = type(iter(''))
```

Comment:
These two symbols have been moved into the `utils` module.

Alternatively, I could make these private so that we don't increase the API surface.

Another thought I had is that we could inline the `(int, float)` tuple into the small handful of call sites where it's checked, but I kept it here since we use the symbol in a few different files.

If we think shrinking the API surface is worth it then I'd be okay inlining it into the calling modules. Let me know what you think about that these.

Reply:
Also, reading through this code made me realize that the code paths that use `ITERATOR_TYPE` might need some attention. It seems like that expression returns a very specific type of iterator, specifically `<class 'str_ascii_iterator'>`, and I suspect that our code was hoping for a common base class of all iterators.

Unrelated to these changes, but just something that came to mind so I figured I'd write it down before I forget. Hope you've been well, cheers.